### PR TITLE
Loosens ActiveSupport to 3

### PR DESCRIPTION
**Pull request by ericschultz**
This is a fix for #289 and a second attempt (original pull request at #290)

The spec_helper now loads the needed hash extensions for ActiveSupport < 4. In ActiveSupport >=4, there's no change other than the ActiveSupport version info being loaded, which has no effect test functioning.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -3,6 +3,12 @@
 require 'simplecov'
 require 'coveralls'
 
+# This works around the hash extensions not being automatically included in ActiveSupport < 4
+require 'active_support/version'
+require 'active_support/core_ext/hash' if ActiveSupport::VERSION &&
+                                          ActiveSupport::VERSION::MAJOR &&
+                                          ActiveSupport::VERSION::MAJOR < 4
```
Comment (LeFnord) [inline spec/spec_helper.rb]: is `ActiveSupport.version.to_s < '4'` the same?

File: grape-entity.gemspec [hunk 2]
```diff
@@ -18,8 +18,7 @@ Gem::Specification.new do |s|
 
   s.rubyforge_project = 'grape-entity'
 
-  s.add_runtime_dependency 'activesupport', '>=4.0'
-  # FIXME: remove dependecy
+  s.add_runtime_dependency 'activesupport', '>= 3.0.0'
```
Comment (LeFnord) [inline grape-entity.gemspec]: please can you change the lines, `FIXME` should be above `multi_json`, thanks

File: spec/spec_helper.rb (hunk 1 above)
Comment (ericschultz) [reply spec/spec_helper.rb]: ActiveSupport 3 has a different API for version (https://github.com/rails/rails/blob/3-2-stable/activesupport/lib/active_support/version.rb). It's closer to 'activesupport/gem_version' in 4+.

File: grape-entity.gemspec (hunk 2 above)
Comment (ericschultz) [reply grape-entity.gemspec]: Will do.

**Comment by LeFnord** [comment]
thanks @ericschultz …

will rebase and mörge it, after this one https://github.com/ruby-grape/grape-entity/pull/301, ok

**Pull request merged**